build.sbt

```diff
@@ -1,12 +1,50 @@
 import scala.language.postfixOps
+import sbt.*
 import scala.sys.process.*
 
-lazy val downloadHarFromZap = taskKey[Unit]("Download HAR from ZAP")
+lazy val publishJourneyTestRecording = taskKey[Unit]("Publish journey test recording")
 
-downloadHarFromZap := {
-  url(
-    s"http://${System.getenv("ZAP_HOST")}/OTHER/exim/other/exportHar/?baseurl=http%3A%2F%2Flocalhost&start=&count="
-  ) #> file("requests.har") !
+publishJourneyTestRecording := {
+  val log                    = streams.value.log
+  val journeyTestProjectName = name.value
+
+  log.info("Attempting to publish journey test recording to artifactory")
+  IO.withTemporaryDirectory { tmp =>
+    log.info("Downloading journey test recording from ZAP")
+    val journeyTestRecording = tmp / s"$journeyTestProjectName.har"
+    url(
+      s"http://${sys.env("ZAP_HOST")}/OTHER/exim/other/exportHar/?baseurl=http%3A%2F%2Flocalhost&start=&count="
+    ) #> journeyTestRecording !!
+
+    log.info("Compressing journey test recording with ZIP")
+    val compressedJourneyTestRecording = tmp / s"$journeyTestProjectName.har.zip"
+    IO.zip(
+      List(journeyTestRecording -> journeyTestRecording.name),
+      compressedJourneyTestRecording,
+      Some(java.time.Instant.now().toEpochMilli)
+    )
+
+    log.info("Uploading journey test recording to artifactory")
+    val username       = sys.env("PLATUI_ARTIFACTORY_USERNAME")
+    val password       = sys.env("PLATUI_ARTIFACTORY_PASSWORD")
```

Comment:
when we do a credential binding in jenkins, we give it (I think) the env vars to place the username and password into - but I'm not totally sure if there's anything extra we need to do to ensure the username and password won't be made available from the injectedEnvVars view in jenkins - something to check with B&D before we merge change to build-jobs